File: CodeForces/Codeforces Round 1065 (Div. 3)/C2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        vector<int> a(n), b(n);
        for (int i = 0; i < n; i++) cin >> a[i];
        for (int i = 0; i < n; i++) cin >> b[i];

        string winner = "Tie";

        for (int k = 20; k >= 0; k--) {
            vector<int> diff_indices;
            for (int i = 0; i < n; i++) {
                int abit = (a[i] >> k) & 1;
                int bbit = (b[i] >> k) & 1;
                if (abit != bbit) {
                    diff_indices.push_back(i + 1); 
                }
            }

            if (diff_indices.size() % 2 == 1) {
                int last = diff_indices.back();
                if (last % 2 == 1) {
                    winner = "Ajisai";
                } else {
                    winner = "Mai";
                }
                break;
            }
        }

        cout << winner << "\n";
    }

    return 0;
}

File: CodeForces/Codeforces Round 1065 (Div. 3)/C2.py
t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

 
    winner = "Tie"

    for k in range(20, -1, -1):
        diff_indices = []
        for i in range(n):
            abit = (a[i] >> k) & 1
            bbit = (b[i] >> k) & 1
            if abit != bbit:
                diff_indices.append(i+1)

        if len(diff_indices) % 2 == 1:   
            last = diff_indices[-1]
            if last % 2 == 1:
                winner = "Ajisai"
            else:
                winner = "Mai"
            break

    print(winner)